tests/fsharp/Compiler/CompilerAssert.fs

```diff
@@ -217,3 +217,23 @@ module CompilerAssert =
                     Assert.AreEqual(expectedErrorMessage, errorMessage)
                 )
         
+    let ParseWithErrors (source: string) expectedParseErrors = 
+        lock gate <| fun () ->
+            let parseResults, _ = checker.ParseAndCheckFileInProject("test.fs", 0, SourceText.ofString source, defaultProjectOptions) |> Async.RunSynchronously
```

Comment:
Is it possible to use `ParseFile` (https://github.com/dotnet/fsharp/blob/master/src/fsharp/service/service.fsi#L113) instead?

Reply:
Just confirming that I think `ParseFile` should be sufficient here, and you could probably get rid of the lock as a result

Reply:
Agree with @cartermp , see if you can use ParseFile.